File: carbon/src/main/java/carbon/widget/ListSearchAdapter.java
package carbon.widget;

import java.util.List;

public class ListSearchAdapter<Type> extends SearchAdapter<Type> {
    private List<Type> data;

    public ListSearchAdapter(List<Type> data) {
        this.data = data;
    }

    @Override
    public int getItemCount() {
        return data.size();
    }

    @Override
    public Type getItem(int i) {
        return data.get(i);
    }

    @Override
    public String[] getItemWords(Type item) {
        return new String[]{item.toString()};
    }
}

File: carbon/src/main/java/carbon/widget/ListSearchAdapter.kt
package carbon.widget

open class ListSearchAdapter<Type> : SearchAdapter<Type> {

    private val data: List<Type>

    constructor(data: List<Type>) : super() {
        this.data = data
    }

    override fun getItemCount(): Int {
        return data.size
    }

    override fun getItem(i: Int): Type {
        return data[i]
    }

    override fun getItemWords(item: Type): Array<String> {
        return arrayOf(item.toString())
    }
}
